ops/manifests/manipulations.py

```diff
@@ -138,10 +148,28 @@ def __call__(self, obj: AnyResource) -> None:
 class Addition(Manipulation):
     """Class used to define objects to add to the original manifests."""
 
-    def __call__(self) -> Optional[AnyResource]:
+    _called: Iterator[AnyResource] = iter(())
+
+    def __call__(self) -> Union[None, AnyResource, Iterable[AnyResource]]:
         """Method called to optionally create an object."""
         ...
 
+    def __iter__(self) -> "Addition":
+        """Treat every addition like a possible collection."""
+        obj = self()
+        if obj is None:
+            self._called = iter(())
+        else:
+            try:
+                self._called = iter(obj)
+            except TypeError:
+                self._called = iter((obj,))
+        return self
+
```

Comment:
I believe I'm missing something here. Why can't we return the iterator directly? Something like:
```suggestion
    def __iter__(self) -> "Addition":
        """Treat every addition like a possible collection."""
        obj = self()
        if obj is None:
            return iter(())
        try:
            return iter(obj)
        except TypeError:
                return iter((obj,))
```

Reply:
b/c mypy didn't like it.  It expects something with an `__iter__` also has a `__next__`

```sh
lint: commands[2]> mypy --config-file tox.ini /home/addyess/git/canonical/ops-lib-manifest/ops/
ops/manifests/manipulations.py:169: error: Incompatible return value type (got "Iterator[Any | Any]", expected "Addition")  [return-value]
ops/manifests/manifest.py:178: error: "Addition" has no attribute "__next__"  [attr-defined]
```

Reply:
OO, i got it.  Thanks for encouraging a path